shared/test/test_assets.rb

```diff
@@ -78,51 +79,80 @@ def test_assets_copy_all
     sound_body = 'stub-sound-contents'
     expected_sound_info = {'filename' =>  sound_filename, 'category' => 'audio', 'size' => sound_body.length}
 
-    put(src_channel_id, image_filename, image_body, 'image/jpeg')
-    put(src_channel_id, sound_filename, sound_body, 'audio/mpeg')
+    put(@assets, src_channel_id, image_filename, image_body, 'image/jpeg')
+    put(@assets, src_channel_id, sound_filename, sound_body, 'audio/mpeg')
     copy_file_infos = JSON.parse(copy_all(src_channel_id, dest_channel_id))
-    dest_file_infos = JSON.parse(list(dest_channel_id))
+    dest_file_infos = JSON.parse(list(@assets, dest_channel_id))
 
     assert_fileinfo_equal(expected_image_info, copy_file_infos[0])
     assert_fileinfo_equal(expected_sound_info, copy_file_infos[1])
     assert_fileinfo_equal(expected_image_info, dest_file_infos[0])
     assert_fileinfo_equal(expected_sound_info, dest_file_infos[1])
 
-    delete(src_channel_id, image_filename)
-    delete(src_channel_id, sound_filename)
-    delete(dest_channel_id, image_filename)
-    delete(dest_channel_id, sound_filename)
-    delete_channel(src_channel_id)
-    delete_channel(dest_channel_id)
+    delete(@assets, src_channel_id, image_filename)
+    delete(@assets, src_channel_id, sound_filename)
+    delete(@assets, dest_channel_id, image_filename)
+    delete(@assets, dest_channel_id, sound_filename)
+    delete_channel(@channels, src_channel_id)
+    delete_channel(@channels, dest_channel_id)
   end
 
-  def test_copy_all_with_no_src
-    copy_all(nil, create_channel)
-    assert @assets.last_response.bad_request?
+  def test_assets_auth
+    owner_channel_id = create_channel(@channels)
+
+    non_owner_assets = Rack::Test::Session.new(Rack::MockSession.new(FilesApi, "studio.code.org"))
+
+    filename = 'dog.jpg'
+    body = 'stub-image-contents'
+    content_type = 'image/jpeg'
+
+    put(@assets, owner_channel_id, filename, body, content_type)
+    assert @assets.last_response.successful?, 'Owner can add a file'
+
+    get(non_owner_assets, owner_channel_id, filename)
+    assert non_owner_assets.last_response.successful?, 'Non-owner can read a file'
+
+    put(non_owner_assets, owner_channel_id, filename, body, content_type)
+    assert !non_owner_assets.last_response.successful?, 'Non-owner cannot write a file'
```

Comment:
changed to `client_error?` so that we are asserting positively rather than negatively. I guess I could go either way on testing specific status codes -- if we decide to change the status code around in the 4xx range around, the important thing is that the client and server code are in sync, which we can't catch in a ruby test.